magellan-sample-advanced/src/main/java/com/wealthfront/magellan/sample/advanced/toolbar/ToolbarHelper.kt

```diff
@@ -1,44 +1,39 @@
 package com.wealthfront.magellan.sample.advanced.toolbar
 
 import android.content.Context
+import androidx.lifecycle.LifecycleObserver
 import com.wealthfront.magellan.Navigator
-import com.wealthfront.magellan.coroutines.ShownLifecycleScope
 import com.wealthfront.magellan.lifecycle.LifecycleAwareComponent
-import com.wealthfront.magellan.lifecycle.attachFieldToLifecycle
-import com.wealthfront.magellan.navigation.NavigationLifecycleEvent
+import com.wealthfront.magellan.navigation.NavigableCompat
+import com.wealthfront.magellan.navigation.NavigationListener
 import com.wealthfront.magellan.navigation.NavigationPropagator
-import kotlinx.coroutines.launch
 
 /**
  * [ToolbarHelper] provides APIs to modify the toolbar for the activity.
  *
  * Ideally, this dependency is provider by dependency injection and configured so that the views are cleaned up when the activity
  * is destroyed with the help of (subcomponents & custom scopes)[https://dagger.dev/dev-guide/subcomponents].
  */
-object ToolbarHelper : LifecycleAwareComponent() {
+object ToolbarHelper : LifecycleAwareComponent(), NavigationListener {
 
-  private val shownScope by attachFieldToLifecycle(ShownLifecycleScope())
   private var toolbarView: ToolbarView? = null
 
   fun init(toolbarView: ToolbarView, navigator: Navigator) {
     this.toolbarView = toolbarView
-    toolbarView.binding.back.setOnClickListener {
-      navigator.goBack()
-    }
+    toolbarView.binding.back.setOnClickListener { navigator.goBack() }
+    NavigationPropagator.addNavigableListener(this)
   }
 
-  override fun onShow(context: Context) {
-    shownScope.launch {
-      NavigationPropagator.events
-        .filterIsInstance<NavigationLifecycleEvent.NavigatedFrom>()
-        .collect { event ->
-          toolbarView?.reset()
-        }
-    }
+  override fun onHide(context: Context) {
+    NavigationPropagator.removeNavigableListener(this)
   }
 
-  override fun onHide(context: Context) {
-    toolbarView = null
+  fun onDestroy() {
+    this.toolbarView = null
```

Comment:
We should probably null this out in `onHide` rather than `onDestroy`

Reply:
I think I had it this way to support the Activity being killed by the OS while backgrounded (as opposed to finished by the back button). But it's just a sample, and this makes the code much simpler